gateway/ln-gateway/src/state_machine/pay.rs

```diff
@@ -328,13 +328,8 @@ impl GatewayPayInvoice {
         debug!("Await payment parameters for outgoing contract {contract_id:?}");
         let account = global_context
             .module_api()
-            .wait_contract(contract_id)
-            .await
-            .map_err(|_| OutgoingPaymentError {
-                contract_id,
-                contract: None,
-                error_type: OutgoingPaymentErrorType::OutgoingContractDoesNotExist { contract_id },
-            })?;
+            .await_contract(contract_id)
+            .await;
```

Comment:
It appears that the previous behavior here was just wrong as an rpc error was interpreted at if the contract did not exist. Seems to me the gateway would then fail to cancel an existing contract.

Reply:
Yes this was likely an oversight, there were no retries here.